File: src/main/java/com/aladdinworksfivefiftyfive/service/IncidentReportService.java
package com.aladdinworksfivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworksfivefiftyfive.domain.IncidentReport;
import com.aladdinworksfivefiftyfive.dto.IncidentReportDTO;
import com.aladdinworksfivefiftyfive.dto.IncidentReportSearchDTO;
import com.aladdinworksfivefiftyfive.dto.IncidentReportPageDTO;
import com.aladdinworksfivefiftyfive.dto.IncidentReportConvertCriteriaDTO;
import com.aladdinworksfivefiftyfive.service.GenericService;
import com.aladdinworksfivefiftyfive.dto.common.RequestDTO;
import com.aladdinworksfivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface IncidentReportService extends GenericService<IncidentReport, Integer> {

	List<IncidentReport> findAll();

	ResultDTO addIncidentReport(IncidentReportDTO incidentReportDTO, RequestDTO requestDTO);

	ResultDTO updateIncidentReport(IncidentReportDTO incidentReportDTO, RequestDTO requestDTO);

    Page<IncidentReport> getAllIncidentReports(Pageable pageable);

    Page<IncidentReport> getAllIncidentReports(Specification<IncidentReport> spec, Pageable pageable);

	ResponseEntity<IncidentReportPageDTO> getIncidentReports(IncidentReportSearchDTO incidentReportSearchDTO);
	
	List<IncidentReportDTO> convertIncidentReportsToIncidentReportDTOs(List<IncidentReport> incidentReports, IncidentReportConvertCriteriaDTO convertCriteria);

	IncidentReportDTO getIncidentReportDTOById(Integer incidentReportId);







}

File: aladdinworksfivefiftyfive_ui/src/services/IncidentReportService.js
import http from "../http-common"; 

class IncidentReportService {
  getAllIncidentReports(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/incidentReport/incidentReports`, searchDTO);
  }

  get(incidentReportId) {
    return this.getRequest(`/incidentReport/${incidentReportId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/incidentReport?field=${matchData}`, null);
  }

  addIncidentReport(data) {
    return http.post("/incidentReport/addIncidentReport", data);
  }

  update(data) {
  	return http.post("/incidentReport/updateIncidentReport", data);
  }
  
  uploadImage(data,incidentReportId) {
  	return http.postForm("/incidentReport/uploadImage/"+incidentReportId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new IncidentReportService();
